### PR TITLE
Check state of app pool

### PR DESCRIPTION
**Pull request by Ajay-MS**

### CODE REVIEW

File: Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/IISWebAppMgmtV2/AppCmdOnTargetMachines.ps1 [hunk 1]
```diff
@@ -148,6 +148,36 @@ function Test-AppPoolExist
...
+    $appCmdPath, $iisVersion = Get-AppCmdLocation -regKeyPath $AppCmdRegKey
+    $appCmdArgs = [string]::Format(' list apppool /name:"{0}"',$appPoolName)
+    $command = "`"$appCmdPath`" $appCmdArgs"
+
+    Write-Verbose "Checking application pool state. Running command : $command"
+
+    $appPoolInfo = Run-Command -command $command -failOnErr $false
+
+    $appPoolName = $appPoolName.Replace('`', '``').Replace('"', '`"').Replace('$', '`$')
+    if($null -ne $appPoolInfo -and $appPoolInfo -like "*`"$appPoolName`"*")
```
Comment (vincentdass) [inline Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/IISWebAppMgmtV2/AppCmdOnTargetMachines.ps1]: Can u explain this?

**Review by vincent1173 — APPROVED** [review]

File: Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/IISWebAppMgmtV2/AppCmdOnTargetMachines.ps1 (hunk 1 above)
Comment (Ajay-MS) [reply Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/IISWebAppMgmtV2/AppCmdOnTargetMachines.ps1]: Discussed offline

File: Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/IISWebAppMgmtV2/AppCmdOnTargetMachines.ps1 [hunk 2]
```diff
@@ -737,6 +767,22 @@ function Start-Stop-Recycle-ApplicationPool {
...
     $command = "`"$appCmdPath`" $appCmdArgs"
 
+    $appPoolState = Get-AppPoolState -appPoolName $appPoolName
+
+    if ($appPoolState)
+    {
+        $isAppPoolInStartedState =  ($appPoolState -like "Started")
+
+        if (($isAppPoolInStartedState -and $action -like "start") -or ((-not $isAppPoolInStartedState) -and $action -like "stop")) {
+            Write-verbose "Application pool '$appPoolName' is already in state of the action '$action'. Hence aborting the operation."
```
Comment (kmkumaran) [inline Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/IISWebAppMgmtV2/AppCmdOnTargetMachines.ps1]: Hence skipping...

File: Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/IISWebAppMgmtV2/AppCmdOnTargetMachines.ps1 [hunk 3]
```diff
@@ -737,6 +767,22 @@ function Start-Stop-Recycle-ApplicationPool {
     $appCmdArgs = [string]::Format('{0} apppool /apppool.name:"{1}"', $action, $appPoolName)
     $command = "`"$appCmdPath`" $appCmdArgs"
 
+    $appPoolState = Get-AppPoolState -appPoolName $appPoolName
+
+    if ($appPoolState)
+    {
+        $isAppPoolInStartedState =  ($appPoolState -like "Started")
+
+        if (($isAppPoolInStartedState -and $action -like "start") -or ((-not $isAppPoolInStartedState) -and $action -like "stop")) {
```
Comment (kmkumaran) [inline Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/IISWebAppMgmtV2/AppCmdOnTargetMachines.ps1]: Is the assumption that not started means "stop" state. Are there other states like "Paused" or "Starting" etc?

File: Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/IISWebAppMgmtV2/AppCmdOnTargetMachines.ps1 [hunk 4]
```diff
@@ -737,6 +767,22 @@ function Start-Stop-Recycle-ApplicationPool {
     $appCmdArgs = [string]::Format('{0} apppool /apppool.name:"{1}"', $action, $appPoolName)
     $command = "`"$appCmdPath`" $appCmdArgs"
 
+    $appPoolState = Get-AppPoolState -appPoolName $appPoolName
+
+    if ($appPoolState)
+    {
+        $isAppPoolInStartedState =  ($appPoolState -like "Started")
```
Comment (kmkumaran) [inline Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/IISWebAppMgmtV2/AppCmdOnTargetMachines.ps1]: Can the states be localized?

File: Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/IISWebAppMgmtV2/AppCmdOnTargetMachines.ps1 (hunk 2 above)
Comment (Ajay-MS) [reply Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/IISWebAppMgmtV2/AppCmdOnTargetMachines.ps1]: Resolved.

File: Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/IISWebAppMgmtV2/AppCmdOnTargetMachines.ps1 (hunk 3 above)
Comment (Ajay-MS) [reply Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/IISWebAppMgmtV2/AppCmdOnTargetMachines.ps1]: There are other possible states. changed logic.

File: Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/IISWebAppMgmtV2/AppCmdOnTargetMachines.ps1 (hunk 4 above)
Comment (Ajay-MS) [reply Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/IISWebAppMgmtV2/AppCmdOnTargetMachines.ps1]: No state won't be localised.

**Review by asranja — APPROVED** [review]

**Pull request merged**